modules/react/collection/lib/useListItemRovingFocus.tsx

```diff
@@ -35,11 +42,17 @@ export const useListItemRovingFocus = createElemPropsHook(useCursorListModel)(
         if (model.state.isVirtualized) {
           model.state.UNSTABLE_virtual.scrollToIndex(item.index);
         }
-        requestAnimationFrame(() => {
-          document
-            .querySelector<HTMLElement>(`[data-focus-id="${`${model.state.id}-${item.id}`}"]`)
-            ?.focus();
-        });
+
+        // In React concurrent mode, there could be several render attempts before the element we're
+        // looking for could be available in the DOM
+        retryEachFrame(() => {
+          const element = document.querySelector<HTMLElement>(
+            `[data-focus-id="${`${model.state.id}-${item.id}`}"]`
+          );
+
+          element?.focus();
+          return !!element;
+        }, 5); // 5 should be enough, right?!
```

Comment:
I ran into an issue with dynamically loaded data in the virtualizer where the DOM was not available in the next animation frame. There is also an issue in React concurrent mode where there may be several render attempts if the render process gets interrupted by other work. React concurrent mode tries to be responsive, but is no longer predictable for timing. We instead retry 5 times over the next 5 frames to see when the DOM is ready for a focus change.